app/javascript/packs/leafletMap.js

```diff
@@ -71,14 +71,45 @@ export async function setupMap() {
   map.on("click", onClick);
 }
 
-function addPin(e, pinNumber) {
-  pins[pinNumber] = L.marker(e.latlng, { icon: pinIcons[pinNumber] });
+function formatQueryParameters(position){
+  return "?long="+position.lng+"&lat="+position.lat
+}
+
+function pinPopupContent(position) {
+  const params = formatQueryParameters(position);
+  return `
+    <ul class=pin-menu>
+      <li><a href="/rooms/new${params}">Add Room</a></li>
+      <li><a href="/buildings/new${params}">Add Building</a></li>
+      <li><a href="/locations/new${params}">Add Location</a></li>
+      <li><a id="deletepin" href="javascript:void(0);">Delete Pin</a></li>
+    </ul>
+  `;
+}
+
+function addPin(position, pinNumber) {
+  pins[pinNumber] = L.marker(position, { icon: pinIcons[pinNumber] });
   pins[pinNumber].addTo(map);
   pins[pinNumber].on("click", function (e) {
-    removePin(pinNumber);
+    pinOnClick(pinNumber, position);
   });
 }
 
+function pinOnClick(pinNumber, position){
+  var popup = L.popup()
+    .setLatLng(position)
+    .setContent(pinPopupContent(position))
+    .openOn(map);
+  document.getElementById("deletepin").addEventListener("click", function(e) {
+    removePinAndPopup(pinNumber, popup);
+  }, false);
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener, the `false` here is the default :)

Reply:
When deleting the `false`, it stops working, so I will keep it :)